src/App.tsx

```diff
@@ -1,56 +1,82 @@
-import React from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
 import './App.scss';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { PeopleList } from './components/PeopleList';
+import { Person } from './types/Person';
+
+function debounce(callback: (...args: string[]) => void, delay: number) {
+  let timerId = 0;
+
+  return (...args: string[]) => {
+    window.clearTimeout(timerId);
+
+    timerId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [person, setPerson] = useState<Person>();
+  const [query, setQuery] = useState('');
+  const [focus, setFocus] = useState(false);
```

Comment:
```suggestion
  const [isFocused, setIsFocused] = useState(false);
```
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig